_posts/2019-10-27-on-code-reviews.md

```diff
@@ -0,0 +1,102 @@
+- **I take the time to try and understand the code.** When I feel like I'm
+having a hard time grasping some of the decisions that have been made, or the
+code itself, I often feel asking about it in writing is difficult. If this
+happens, I either take a step away from the computer and try again later, or ask
+to go over it in person (face-to-face or in a video call, _not_ a phone call.
+Being able to see each other aids understanding).<br />
+An exception to this would be a pull request I feel is introducing too many
+concepts at once, in which case I'll either ask if we could break up the pull
+request in more bite-sized pieces, offering to help do so in person if need be.
+- **I ask questions.** I would call myself a curious person. If I see something
+I don't know about, or want to know more about (e.g. a piece of code, business
+logic), I ask. If the review already has many comments, I might do this
+outside the review itself.<br />
+Also, if I see code that appears to be copied, where I feel improvements can be
+made, I ask if this is something that has been considered/if this is something
+for us to consider in a follow-up pull request.
+- **I re-read a review before I submit it.** I don't always do this, but I
+know it helps. I've seen the author ask about comments I wrote, not (fully)
+understanding what I meant. Though communication is hard, I feel this has
+been happening less over time.
+- **I feel responsible for the code I review,** meaning I use "we", not
+"you". "We could consider adding a comment explaining why we need to ignore X",
+not "you".
+- **I review my own code.** Especially in larger pull requests, I look through
+the code before assigning a reviewer, asking open questions to the reviewer,
+making notes and/or making some changes. Seeing the "bigger" picture often helps
+me catch issues/improvements to be made early, hopefully making the code review
+more enjoyable in the process.
+- **I don't use words like "easy", "simply", or similar.** The fact something
+would be any of those is always an assumption to whoever is reading it, and
+serves no purpose.
+
+---
+
+Keep in mind that this is not an exhaustive list, but rather a bunch of the
+principles I try to keep in mind while reviewing, and there's still so much to
+learn. If you have any thoughts on this post, like guidelines that you're using
+that I've missed here, I'd love to hear about them!
+
+Bonus: I'm a big fan of a tool we built at work to get a random reviewer for
+your code. It's called `lotto`, and picks a random reviewer outside of your
+team. It's great.
+
+## On Non-Code Reviews
+
+Oh, and most of this also applies reviewing other things, like blog posts and
+documentation. For those in particular, I prefer printing the documents (so it
+is easy to make both corrections and remarks). That also giving me the
+opportunity to find another place to work, away from my computer. Natural light
+helps!
+
+If possible, I always discuss it in person afterwards.
```

Comment:
```suggestion
If possible, I always discuss it in person afterward.
```

Afterwards is a non American spelling (according to Grammarly)

Reply:
It indeed (somewhat) [seems like that](https://grammarist.com/usage/afterward-afterwards/). Today I learned.